Yosemite/Yosemite/Stores/ShipmentStore.swift

```diff
@@ -0,0 +1,90 @@
+import Foundation
+import Networking
+import Storage
+import CocoaLumberjack
+
+
+// MARK: - ShipmentStore
+//
+public class ShipmentStore: Store {
+
+    /// Shared private StorageType for use during then entire ShipmentStore sync process
+    ///
+    private lazy var sharedDerivedStorage: StorageType = {
+        return storageManager.newDerivedStorage()
+    }()
+
+    /// Registers for supported Actions.
+    ///
+    override public func registerSupportedActions(in dispatcher: Dispatcher) {
+        dispatcher.register(processor: self, for: ShipmentAction.self)
+    }
+
+    /// Receives and executes Actions.
+    ///
+    override public func onAction(_ action: Action) {
+        guard let action = action as? ShipmentAction else {
+            assertionFailure("ShipmentStore received an unsupported action")
+            return
+        }
+
+        switch action {
+        case .synchronizeShipmentTrackingData(let siteID, let orderID, let onCompletion):
+            synchronizeShipmentTrackingData(siteID: siteID, orderID: orderID, onCompletion: onCompletion)
+        }
+    }
+}
+
+
+// MARK: - Services!
+//
+private extension ShipmentStore {
+
+    func synchronizeShipmentTrackingData(siteID: Int, orderID: Int, onCompletion: @escaping (Error?) -> Void) {
+        let remote = ShipmentsRemote(network: network)
+        remote.loadShipmentTrackings(for: siteID, orderID: orderID) { [weak self] (shipmentTrackingData, error) in
+            guard let readOnlyShipmentTrackingData = shipmentTrackingData else {
+                onCompletion(error)
+                return
+            }
+
+            self?.upsertShipmentTrackingDataInBackground(siteID: siteID, orderID: orderID, readOnlyShipmentTrackingData: readOnlyShipmentTrackingData) {
+                onCompletion(nil)
+            }
+        }
+    }
+}
+
+
+// MARK: - Persistence
+//
+extension ShipmentStore {
```

Comment:
I was wondering, should this extension be private was well?

Reply:
@ctarda we typically leave these `*Store` persistence extensions as internal/public so they are directly accessible to unit tests...this is a trade off I realize.